source/extensions/access_loggers/fluentd/fluentd_access_log_impl.cc

```diff
@@ -122,7 +122,7 @@ FluentdAccessLoggerCacheImpl::getOrCreateLogger(const FluentdAccessLogConfigShar
   const auto cache_key = MessageUtil::hash(*config);
   const auto it = cache.access_loggers_.find(cache_key);
   if (it != cache.access_loggers_.end()) {
-    return it->second;
+    return it->second.expired() ? nullptr : it->second.lock();
```

Comment:
If the exist entry was expired, you need to recreate one. 

(nit: And it would be better to cleanup all the expired entry.